iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -435,6 +486,70 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
             }
         }
 
+        private void StartLoopAsync(DateTime refreshOn, CancellationToken cancellationToken)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, nameof(StartLoopAsync));
+
+            if (_loopCancellationTokenSource == null
+                || _refreshLoop == null)
+            {
+                (this as IAmqpAuthenticationRefresher)?.StopLoop();
+            }
+
+            _loopCancellationTokenSource = new CancellationTokenSource();
```

Comment:
If this previously held a value, it did not get disposed here when it was overwritten.